### DISCUSSION TITLE
Eglot over tramp with external methods

### DISCUSSION THREAD

**Discussion by kohnish** [joaotavora/eglot#1294] [Q&A] [answered]
At the moment, eglot support for tramp is limited to ssh.  
Would it be possible to support external methods like sftp and sshfs, while letting language servers run on the (powerful) remote machine?

**Comment by joaotavora** [comment]
> At the moment, eglot support for tramp is limited to ssh.

I personally haven't tried any other method, but as far as I can read from the code, there is no explicit invocation of the `ssh` program.  What happens if you visit a source file via some other tramp method (`sftp` or `sshfs`) on a server where the language server is installed and press `M-x eglot`?

**Reply by kohnish** [reply]
It starts clangd but without stty command  if I check the process from OS and eglot is not functional over sshfs.  
  
Another thing I noticed is that (vc-root-dir) returns nil over sshfs.  
  
I'll check sftp later.

**Reply by kohnish** [reply]
With sftp eglit doesn't find remote clangd executable no matter what. 

According to the documentation only some features like shell.el, compile.el support gvfs backend. 
  
https://www.gnu.org/software/emacs/manual/html_node/tramp/Remote-processes.html

**Reply by joaotavora** [reply, marked as answer]
I'm not a TRAMP expert, so I can't speak to that.  My observation is only that Eglot doesn't have any special preference for `ssh`, not that I can detect at least.  So any feature requests or inquiries related to alternate methods of starting processes remotely should probably be directed to bug-gnu-emacs@gnu.org, CC-ing me and Michael Albinus in the email.  IOW, it's fine to use Eglot as a test bed, but I don't think this is an Eglot problem.

**Reply by kohnish** [reply]
Will do. Thanks for the quick support!

**Reply by joaotavora** [reply]
Thanks. Moved to bug#66093
